interface.go

```diff
@@ -0,0 +1,34 @@
+package event
+
+type SubSettings struct {}
+type SubOption func(*SubSettings)
+
+type EmitterSettings struct {}
+type EmitterOption func(*EmitterSettings)
+
+type Bus interface {
+	// Subscribe creates new subscription. Failing to drain the incoming channel
+	// will cause publishers to get blocked
+	//
+	// evtTypes only accepts typed nil pointers, and uses the type information to
+	// select output type
+	//
+	// Example:
+	// sub, cancel, err := eventbus.Subscribe(new(os.Signal))
+	// defer cancel()
+	//
+	// evt := (<-sub).(os.Signal) // guaranteed to be safe
+	Subscribe(eventType interface{}, opts ...SubOption) (<-chan interface{}, CancelFunc, error)
```

Comment:
also consider string events with a namespace i.e. `("autonat", "status_changed")` which could be rather simply stored in a map

Reply:
ah, i see you're using reflect to get equality. that works with me.